src/components/NewCommentForm/NewCommentForm.jsx

```diff
@@ -1,39 +1,77 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './NewCommentForm.scss';
+import PropTypes from 'prop-types';
 
-export const NewCommentForm = () => (
-  <form className="NewCommentForm">
-    <div className="form-field">
-      <input
-        type="text"
-        name="name"
-        placeholder="Your name"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <input
-        type="text"
-        name="email"
-        placeholder="Your email"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <textarea
-        name="body"
-        placeholder="Type comment here"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <button
-      type="submit"
-      className="NewCommentForm__submit-button button"
-    >
-      Add a comment
-    </button>
-  </form>
-);
+export const NewCommentForm = ({ postId, onAdd }) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [body, setBody] = useState('');
+
+  const inputHandler = (event, callback) => {
+    callback(event.target.value);
+  };
+
+  const submitHandler = (event) => {
+    event.preventDefault();
+
+    onAdd({
+      name,
+      body,
+      email,
+      postId,
+    });
+
+    clearForm();
+  };
+
+  const clearForm = () => {
+    setBody('');
+    setName('');
+    setEmail('');
+  };
+
+  return (
+    <form className="NewCommentForm" onSubmit={submitHandler}>
+      <div className="form-field">
+        <input
+          type="text"
+          placeholder="Your name"
+          className="NewCommentForm__input"
+          value={name}
+          onChange={event => inputHandler(event, setName)}
```

Comment:
ох как тут легко запутаться!
```suggestion
          onChange={event => setName(event.target.value)}
```
Намного понятнее